pkg/resources/metrics.go

```diff
@@ -0,0 +1,60 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	prometheusv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	croResources "github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreatePostgresConnectivityAlert creates a PrometheusRule alert to watch for the connectivity
+// of a Postgres instance
+func CreatePostgresConnectivityAlert(ctx context.Context, client k8sclient.Client, cr *v1alpha1.Postgres) (*prometheusv1.PrometheusRule, error) {
+	productName := cr.Labels["productName"]
+	alertName := productName + "PostgresConnectionFailed"
+	ruleName := fmt.Sprintf("connectivity-rule-%s", cr.Name)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("absent(%s{exported_namespace='%s',resourceID='%s',productName='%s'} == 1)",
+			croResources.DefaultPostgresConnectionMetricName, cr.Namespace, cr.Name, productName),
+	)
+	alertDescription := fmt.Sprintf("Unable to connect to Postgres instance: '%s' (strategy: %s) for product: %s", cr.Name, cr.Status.Strategy, productName)
+	labels := map[string]string{
+		"severity":    "critical",
+		"productName": cr.Labels["productName"],
+	}
+
+	// create the rule
+	pr, err := croResources.ReconcilePrometheusRule(ctx, client, ruleName, cr.Namespace, alertName, alertDescription, alertExp, labels)
+	if err != nil {
+		return nil, err
+	}
+	return pr, nil
+}
+
+// CreateRedisConnectivityAlert creates a PrometheusRule alert to watch for the connectivity
+// of a Redis cache
+func CreateRedisConnectivityAlert(ctx context.Context, client k8sclient.Client, cr *v1alpha1.Redis) (*prometheusv1.PrometheusRule, error) {
+	productName := cr.Labels["productName"]
+	alertName := productName + "RedisCacheConnectionFailed"
+	ruleName := fmt.Sprintf("connectivity-rule-%s", cr.Name)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("absent(%s{exported_namespace='%s',resourceID='%s',productName='%s'} == 1)",
+			croResources.DefaultRedisConnectionMetricName, cr.Namespace, cr.Name, productName),
+	)
+	alertDescription := fmt.Sprintf("Unable to connect to Redis instance: '%s' (strategy: %s) for the product: %s", cr.Name, cr.Status.Strategy, productName)
```

Comment:
@ciaranRoche might be worth including the namespace & resourceID in the alert message to help debugging